map_gen/shared/generate.lua

```
@@ -312,6 +314,7 @@ function Public.register()
 end
 
 --- Returns the surfaces that the generate functions will act on
+-- Warning! Changing this table after on_init or on_load has run will cause desyncs!
```

Let's throw an error if someone calls this at runtime